src/latency.h

```diff
@@ -89,4 +91,34 @@ void latencyAddSample(const char *event, mstime_t latency);
 #define latencyRemoveNestedEvent(event_var,nested_var) \
     event_var += nested_var;
 
+struct durationStats {
+    unsigned long long cnt;
+    long long sum;
+    long long max;
+};
+
+void durationAddSample(const char *event, ustime_t duration, int add_to_latency);
+
+/* Record the start time of an event. In order to avoid performance implication due to 
+ * querying the clock using a system call every time we measure a duration, we use a 
+ * monotonic clock, when we are sure its cost is very low, and fall back to non-monotonic 
+ * call otherwise. */
+#define durationStartMonitor(var)                   \
+    if (monotonicGetType() == MONOTONIC_CLOCK_HW) { \
+        var = getMonotonicUs();                     \
+    } else {                                        \
+        var = ustime();                             \
+    }
+
+/* End monitoring an event, compute the difference with the current time to check the 
+ * amount of time elapsed. */
+#define durationEndMonitor(var)                     \
+    if (monotonicGetType() == MONOTONIC_CLOCK_HW) { \
+        var = getMonotonicUs() - var;               \
+    } else {                                        \
+        var = ustime() - var;                       \
+    }
```

Comment:
the monotonic clock itself already has a fallback to clock_gettime (which IIRC is a system call similar if not faster than gettimeofday), why do we need the ustime fallback?